src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5850,9 +5871,9 @@ public override BoundNode VisitAttribute(BoundAttribute node)
             return null;
         }
 
-        public override BoundNode VisitDeconstructValuePlaceholder(BoundDeconstructValuePlaceholder node)
+        public override BoundNode VisitExpressionWithNullability(BoundExpressionWithNullability node)
         {
-            SetNotNullResult(node);
+            ResultType = TypeWithState.Create(node.Type, node.NullableAnnotation == NullableAnnotation.Annotated ? NullableFlowState.MaybeNull : NullableFlowState.NotNull);
```

Comment:
Feels like there may be a bug here. Probably should be `TypeWithAnnotation.Create(node.Type, node.NullableAnnotation).ToTypeWithState()` instead. `ToTypeWithState` handles type parameters.
Please confirm and add a test to illustrate if applicable.